### ISSUE TITLE
/usr/local/bin/unsquashfs: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by ctroupin** [apptainer/singularity#6139]
### Version of Singularity:

What version of Singularity are you using? Run:

```
singularity version 3.8.1
```

### Expected behavior

By running 
```bash
$ sudo singularity build lolcow.sif library://sylabs-jms/testing/lolcow
```
taken from the [doc](https://singularity.hpcng.org/user-docs/3.8/build_a_container.html) I expected the container to be build.

### Actual behavior

```
INFO:    Starting build...
INFO:    Using cached image
INFO:    Verifying bootstrap image /root/.singularity/cache/library/sha256.5022b5e7c7249c40119a875c1ace0700ced4099e077acc75d0132190254563a4
WARNING: integrity: signature not found for object group 1
WARNING: Bootstrap image could not be verified, but build will continue.
ERROR:   unpackSIF failed: root filesystem extraction failed: extract command failed: WARNING: Skipping mount /etc/hosts [binds]: /etc/hosts doesn't exist in container
WARNING: Skipping mount /etc/localtime [binds]: /etc/localtime doesn't exist in container
WARNING: Skipping mount proc [kernel]: /proc doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/tmp [tmp]: /tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/var/tmp [tmp]: /var/tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
/usr/local/bin/unsquashfs: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
: exit status 127
FATAL:   While performing build: packer failed to pack: root filesystem extraction failed: extract command failed: WARNING: Skipping mount /etc/hosts [binds]: /etc/hosts doesn't exist in container
WARNING: Skipping mount /etc/localtime [binds]: /etc/localtime doesn't exist in container
WARNING: Skipping mount proc [kernel]: /proc doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/tmp [tmp]: /tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/var/tmp [tmp]: /var/tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
/usr/local/bin/unsquashfs: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
: exit status 12y
```

The library `libz.so.1` is present (many times) on my machine.
```
$ ldd /usr/local/bin/unsquashfs
	linux-vdso.so.1 (0x00007ffcb9163000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f54ed951000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f54ed802000)
	libz.so.1 => /usr/local/lib/libz.so.1 (0x00007f54ed7e5000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f54ed5f3000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f54edb42000)
```
I recently updated `squashfs`
```
$ /usr/local/bin/unsquashfs -v
unsquashfs version 4.5 (2021/07/22)
```
but that did not help. 

### Steps to reproduce this behavior

```
$ sudo singularity build lolcow.sif library://sylabs-jms/testing/lolcow
```

### What OS/distro are you running

```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
I previously tried with Ubuntu 16, which gave the exact same error message as above.

### How did you install Singularity

Following this: https://singularity.hpcng.org/user-docs/3.8/quick_start.html
```
$ ./mconfig && \
    make -C builddir && \
    sudo make -C builddir install
```

### What I tried...

- [x] `sudo apt-get install zlib1g`; the library was already installed
- [x] `( unset LD_LIBRARY_PATH ; sudo singularity build lolcow.sif library://sylabs-jms/testing/lolcow )`
(following issue https://github.com/hpcng/singularity/issues/5666)
- [x] Changing `LD_LIBRARY_PATH` to use another location for `libz.so.1`.

**Comment by DrDaveD** [comment]
The error is different but it looks like it might be related to #6113.  Could you please check to see if you can learn anything from that?  Sylabs just made a fix for that issue and released it in a singularity-ce 3.8.2 yesterday.  It would be good if you could check if that fixes it, because the fix will also be in hpcng/singularity 3.8.2 when that comes out.

I don't have an Ubuntu machine, and your command works for me on centos7 so I can't reproduce the problem.

**Comment by ctroupin** [comment]
Thanks. I had seen that related issue discussion but forgot to mention it in my message.

I'll try that, in fact I've tried `singularity-ce 3.8.2` as well but the issue was the same. Anyway, I'll double check and will report later on. Another colleague could not reproduce the problem with Ubuntu 18.

**Comment by ctroupin** [comment]
I confirm I'm having the same error with
```
$ /usr/local/bin/singularity --version
singularity-ce version 3.8.2
```
Also tested with 
- [x] unsquashfs version 4.5 (2021/07/22)
- [x] unsquashfs version 4.4 (2019/08/29)

I'm wondering if the problem could somehow come from `LD_LIBRARY_PATH`. I've different tries (unsetting it, changing the list of paths), nothing successful so far.

**Comment by DrDaveD** [comment]
Since you've shown it also fails on the latest singularity-ce I suggest you make another issue over at https://github.com/sylabs/singularity and link to this one.  They may have better luck at reproducing it and finding a fix.

**Comment by DrDaveD** [comment]
Sylabs addressed this in their [issue #273](https://github.com/sylabs/singularity/issues/273).  Their solution in [pr #272](https://github.com/sylabs/singularity/pull/272) seems to match what we had already discuss doing in hpcng pr #5859 and its corresponding issue #5858.

**Comment by ctroupin** [comment]
Thanks a lot for the help, I'm glad I now have something working!
